components/Accordion/react/Accordion.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react'
+import clsx from 'clsx'
+import { CssClasses } from '../constants'
+import type { AccordionProps } from '../constants'
+import { DetailsAnimation } from '@cypress-design/details-animation'
+import { IconChevronDownSmall } from '@cypress-design/react-icon'
+
+export interface AccordionPropsReact extends AccordionProps {
+  icon?: React.FC<React.SVGProps<SVGSVGElement>>
+}
+
+export const Accordion: React.FC<
+  AccordionPropsReact & React.HTMLProps<HTMLDetailsElement>
+> = ({
+  title,
+  description,
+  icon: Icon,
+  separator = false,
+  children,
+  headingClassName,
+  titleClassName,
+  descriptionClassName,
+  fullWidthContent,
+  ...rest
+}) => {
+  const details = React.useRef(null)
+  const content = React.useRef(null)
+  React.useEffect(() => {
+    if (details.current && content.current) {
+      new DetailsAnimation(details.current, content.current)
+    }
+  }, [])
+  return (
+    <details {...rest} className={clsx(rest.className)} ref={details}>
+      <summary
+        className={clsx(
+          'flex items-center',
+          CssClasses.summary,
+          headingClassName ?? CssClasses.summaryColor
+        )}
+      >
+        {Icon && <Icon className={CssClasses.icon} />}
+        {Icon && separator && <hr className={CssClasses.separator} />}
+        <div className="flex-grow pr-16px">
+          <div
+            className={clsx(
+              titleClassName ?? CssClasses.summaryTitleColor,
+              CssClasses.summaryTitle
+            )}
+          >
+            {title}
+          </div>
+          {description && (
+            <div
+              className={clsx(
+                CssClasses.summaryDescription,
+                descriptionClassName ?? CssClasses.summaryDescriptionColor
+              )}
+            >
+              {description}
+            </div>
+          )}
+        </div>
+        <IconChevronDownSmall
+          strokeColor="gray-300"
+          className={clsx('open:icon-dark-gray-500', CssClasses.chevron)}
+        />
+      </summary>
+      <div ref={content} className={CssClasses.contentWrapper}>
+        {fullWidthContent ? (
+          <>{children}</>
+        ) : (
+          <div className={CssClasses.content}>{children}</div>
```

Comment:
Are you saying I should replace the `<div>` with a `<p>`?

Reply:
And create base styles for `<p>`?

Reply:
No, I was saying the children would be `p`s. It definitely shouldn't be a single p, because there are times there are multiple paragraphs in here. 

Anyway, there's no need to do anything in here right now, I guess. Based on the convo we had with Ryan.

Reply:
Yes, it sounds like we do need base styles for `<p>`. That doesn't have to be in this PR, though.

Reply:
💪 👍